client/tracing/src/lib.rs

```diff
@@ -291,12 +303,17 @@ impl Subscriber for ProfilingSubscriber {
 		let id = Id::from_u64(self.next_id.fetch_add(1, Ordering::Relaxed));
 		let mut values = Values::default();
 		attrs.record(&mut values);
-		// If this is a wasm trace, check if target/level is enabled
-		if let Some(wasm_target) = values.string_values.get(WASM_TARGET_KEY) {
-			if !self.check_target(wasm_target, attrs.metadata().level()) {
-				return id
+
+		if attrs.metadata().target() == WASM_TRACE_IDENTIFIER {
+			// If this is a wasm trace, check if target/level is enabled
+			if let Some(wasm_target) = values.string_values.get(WASM_TARGET_KEY) {
+				if !self.check_target(wasm_target, attrs.metadata().level()) {
+					// returning as disabled
+					return Id::from_u64(0)
```

Comment:
Won't this just panic instead of returning 0?

https://docs.rs/tracing/0.1.19/tracing/span/struct.Id.html#method.from_u64

Reply:
hmm.. yeah, well spotted. Interestingly I for sure ran it without all trace-levels activated and it didn't panic... :thinking:  ... but in general this means the given approach can't work. I'll refactor this part.